--- git status ---
On branch main
Your branch is up to date with 'origin/main'.

Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	deleted:    logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/model_0.pt
	deleted:    logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/model_1000.pt
	deleted:    logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/model_2000.pt
	deleted:    logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/model_3000.pt
	deleted:    logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-0.mp4
	deleted:    logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-10000.mp4
	deleted:    logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-20000.mp4
	deleted:    logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-30000.mp4
	deleted:    logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-40000.mp4
	deleted:    logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-50000.mp4
	deleted:    logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-60000.mp4
	deleted:    logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-70000.mp4
	deleted:    logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-80000.mp4
	modified:   source/pm01_dirct/pm01_dirct/tasks/direct/pm01_dirct/pm01_dirct_env.py
	modified:   source/pm01_dirct/pm01_dirct/tasks/direct/pm01_dirct/pm01_dirct_env_cfg.py

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/exported/
	logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/param
	logs/rsl_rl/pm01_flat/2026-03-31_09-55-41/
	logs/rsl_rl/pm01_flat/2026-03-31_11-33-55/
	logs/rsl_rl/pm01_flat/2026-03-31_13-28-36/
	logs/rsl_rl/pm01_flat/2026-03-31_14-29-04/

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/model_0.pt b/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/model_0.pt
deleted file mode 100644
index d1950b7..0000000
--- a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/model_0.pt
+++ /dev/null
@@ -1,3 +0,0 @@
-version https://git-lfs.github.com/spec/v1
-oid sha256:70810b0a9ead0adf68c94890a77e4605abb8c2e9a4f2ca75bbdd30fdc1d8b1cc
-size 1121879
diff --git a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/model_1000.pt b/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/model_1000.pt
deleted file mode 100644
index 49d1063..0000000
--- a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/model_1000.pt
+++ /dev/null
@@ -1,3 +0,0 @@
-version https://git-lfs.github.com/spec/v1
-oid sha256:3dd82838fe22425c47893fe74a15e50554e4eb1e8a5161d807e6d5778ee76d76
-size 1122549
diff --git a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/model_2000.pt b/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/model_2000.pt
deleted file mode 100644
index c8880bf..0000000
--- a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/model_2000.pt
+++ /dev/null
@@ -1,3 +0,0 @@
-version https://git-lfs.github.com/spec/v1
-oid sha256:b658f7551003ab911dd2c435ee1cba56eb90070a17173f3dcb93f1c4abdd0850
-size 1122549
diff --git a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/model_3000.pt b/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/model_3000.pt
deleted file mode 100644
index ecaf2b0..0000000
--- a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/model_3000.pt
+++ /dev/null
@@ -1,3 +0,0 @@
-version https://git-lfs.github.com/spec/v1
-oid sha256:69f8a0b916a2030689d1c3a823ea9c003dfea0ae54d4d80664c71038a3ff8cc6
-size 1122549
diff --git a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-0.mp4 b/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-0.mp4
deleted file mode 100644
index e388b5c..0000000
--- a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-0.mp4
+++ /dev/null
@@ -1,3 +0,0 @@
-version https://git-lfs.github.com/spec/v1
-oid sha256:812c59d9f5de85923f0839a13816cc0b7b115c64f80f5c7a92e58acb310c411b
-size 865417
diff --git a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-10000.mp4 b/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-10000.mp4
deleted file mode 100644
index 9d0cee7..0000000
--- a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-10000.mp4
+++ /dev/null
@@ -1,3 +0,0 @@
-version https://git-lfs.github.com/spec/v1
-oid sha256:3de0a677d40e666997920b671d7665f144a18c69a5a349f8ba6bcf0f2719e8d0
-size 711590
diff --git a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-20000.mp4 b/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-20000.mp4
deleted file mode 100644
index 4aceee7..0000000
--- a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-20000.mp4
+++ /dev/null
@@ -1,3 +0,0 @@
-version https://git-lfs.github.com/spec/v1
-oid sha256:ee13ce7cf4501120fb6501e79d809d71a6b91986e1a84e6611cff20f29db9f81
-size 544719
diff --git a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-30000.mp4 b/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-30000.mp4
deleted file mode 100644
index ccab0c1..0000000
--- a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-30000.mp4
+++ /dev/null
@@ -1,3 +0,0 @@
-version https://git-lfs.github.com/spec/v1
-oid sha256:a5dbbfc700f2807aa3cb9da636d5dd360422bf23238ab6b87234b342d352a48c
-size 587749
diff --git a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-40000.mp4 b/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-40000.mp4
deleted file mode 100644
index 471f81d..0000000
--- a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-40000.mp4
+++ /dev/null
@@ -1,3 +0,0 @@
-version https://git-lfs.github.com/spec/v1
-oid sha256:ee066846f4cc8005d651bf8af99f7ce2ef4bb71e484d1ee24bcef5b981fd7c46
-size 579489
diff --git a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-50000.mp4 b/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-50000.mp4
deleted file mode 100644
index 6c866d7..0000000
--- a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-50000.mp4
+++ /dev/null
@@ -1,3 +0,0 @@
-version https://git-lfs.github.com/spec/v1
-oid sha256:ae091cd2d7ab20aac1bf093854f47050c644395107ad7dc038ea2f8b5b07cd2c
-size 637646
diff --git a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-60000.mp4 b/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-60000.mp4
deleted file mode 100644
index 7ff23e4..0000000
--- a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-60000.mp4
+++ /dev/null
@@ -1,3 +0,0 @@
-version https://git-lfs.github.com/spec/v1
-oid sha256:adb8d544ab46e6d1bc1aba2839c10e14ab2ab10660c7c870d7ede7b9507d0e96
-size 595544
diff --git a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-70000.mp4 b/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-70000.mp4
deleted file mode 100644
index 9128842..0000000
--- a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-70000.mp4
+++ /dev/null
@@ -1,3 +0,0 @@
-version https://git-lfs.github.com/spec/v1
-oid sha256:3bdebeea35092f4bafbd887f05c48627d5b7c983b2c18f5507f3db0971ac0918
-size 681964
diff --git a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-80000.mp4 b/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-80000.mp4
deleted file mode 100644
index db35b4a..0000000
--- a/logs/rsl_rl/pm01_flat/2026-03-30_15-18-36/videos/train/rl-video-step-80000.mp4
+++ /dev/null
@@ -1,3 +0,0 @@
-version https://git-lfs.github.com/spec/v1
-oid sha256:23a507fd28effe442aafc5632c2337b22d69305863e74a34435407e57fcd80e6
-size 565493
diff --git a/source/pm01_dirct/pm01_dirct/tasks/direct/pm01_dirct/pm01_dirct_env.py b/source/pm01_dirct/pm01_dirct/tasks/direct/pm01_dirct/pm01_dirct_env.py
index 00a53a7..6d2dc70 100644
--- a/source/pm01_dirct/pm01_dirct/tasks/direct/pm01_dirct/pm01_dirct_env.py
+++ b/source/pm01_dirct/pm01_dirct/tasks/direct/pm01_dirct/pm01_dirct_env.py
@@ -148,7 +148,8 @@ class Pm01DirctEnv(DirectRLEnv):
         first_contact = self._contact_sensor.compute_first_contact(self.step_dt)[:, self._feet_ids]
         last_air_time = self._contact_sensor.data.last_air_time[:, self._feet_ids]
         air_time = torch.sum((last_air_time - self.cfg.feet_air_time_threshold) * first_contact, dim=1) * (
-            torch.norm(self._commands[:, :2], dim=1) > 0.1
+            # torch.norm(self._commands[:, :2], dim=1) > 0.1
+            torch.norm(self._commands[:, :3], dim=1) > 0.1
         )
         # undesired contacts
         # net_contact_forces = self._contact_sensor.data.net_forces_w_history
@@ -186,6 +187,7 @@ class Pm01DirctEnv(DirectRLEnv):
         #stand_still
         angle = asset.data.joint_pos - asset.data.default_joint_pos
         joint_deviation_l1 = torch.sum(torch.abs(angle), dim=1)
+        # stand_still = joint_deviation_l1 * (torch.norm(self._commands[:, :2], dim=1) < self.cfg.command_threshold)
         stand_still = joint_deviation_l1 * (torch.norm(self._commands[:, :3], dim=1) < self.cfg.command_threshold)
         
         rewards = {
diff --git a/source/pm01_dirct/pm01_dirct/tasks/direct/pm01_dirct/pm01_dirct_env_cfg.py b/source/pm01_dirct/pm01_dirct/tasks/direct/pm01_dirct/pm01_dirct_env_cfg.py
index bf62179..f8ab4a0 100644
--- a/source/pm01_dirct/pm01_dirct/tasks/direct/pm01_dirct/pm01_dirct_env_cfg.py
+++ b/source/pm01_dirct/pm01_dirct/tasks/direct/pm01_dirct/pm01_dirct_env_cfg.py
@@ -165,7 +165,7 @@ class Pm01DirctFaltEnvCfg(DirectRLEnvCfg):
     feet_air_time_threshold = 0.5
 
     track_lin_vel_xy_reward_scale = 1.0
-    track_ang_vel__reward_scale = 1.0   #0.5
+    track_ang_vel__reward_scale = 2.0   #1.0
     lin_vel_z_l2_reward_scale = -2.0
     ang_vel_reward_scale = -0.05
     joint_torque_reward_scale = -2e-5
@@ -178,4 +178,4 @@ class Pm01DirctFaltEnvCfg(DirectRLEnvCfg):
     joint_deviation_knee_reward_scale = -0.01
     joint_deviation_arms_reward_scale = -0.1
     joint_deviation_torso_reward_scale = -0.01
-    stand_still = 0.0
+    stand_still = -3.0