--- git status ---
On branch main
Your branch is up to date with 'origin/main'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   note.md
	modified:   source/pm01_dirct/pm01_dirct/tasks/direct/pm01_dirct/agents/rsl_rl_ppo_cfg.py
	modified:   source/pm01_dirct/pm01_dirct/tasks/direct/pm01_dirct/pm01_dirct_env_cfg.py

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	logs/rsl_rl/pm01_flat/2026-03-27_13-15-19/ param
	logs/rsl_rl/pm01_flat/2026-03-30_11-12-51/

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/note.md b/note.md
index 3799492..1756789 100644
--- a/note.md
+++ b/note.md
@@ -6,7 +6,7 @@ conda activate isaaclab2.3.2
 source /home/ubuntu/workspace/isaac/isaac-sim/setup_conda_env.sh
 
 # 训练与测试
-python3 scripts/rsl_rl/train.py --task=Dirct-Pm01-v0 --num_envs=4096 --headless --video --video_length 200 --video_interval 10000 --max_iterations 6000
+python3 scripts/rsl_rl/train.py --task=Dirct-Pm01-v0 --num_envs=4096 --headless --video --video_length 200 --video_interval 10000 --max_iterations 5000
 
 python3 scripts/rsl_rl/play.py --task=Dirct-Pm01-v0 --num_envs=8 --checkpoint 
 
diff --git a/source/pm01_dirct/pm01_dirct/tasks/direct/pm01_dirct/agents/rsl_rl_ppo_cfg.py b/source/pm01_dirct/pm01_dirct/tasks/direct/pm01_dirct/agents/rsl_rl_ppo_cfg.py
index be0299c..53526c2 100644
--- a/source/pm01_dirct/pm01_dirct/tasks/direct/pm01_dirct/agents/rsl_rl_ppo_cfg.py
+++ b/source/pm01_dirct/pm01_dirct/tasks/direct/pm01_dirct/agents/rsl_rl_ppo_cfg.py
@@ -11,8 +11,8 @@ from isaaclab_rl.rsl_rl import RslRlOnPolicyRunnerCfg, RslRlPpoActorCriticCfg, R
 @configclass
 class PM01RoughPPORunnerCfg(RslRlOnPolicyRunnerCfg):
     num_steps_per_env = 24
-    max_iterations = 3000
-    save_interval = 400
+    max_iterations = 5000
+    save_interval = 1000
     experiment_name = "pm01_rough"
     policy = RslRlPpoActorCriticCfg(
         init_noise_std=1.0,
@@ -43,7 +43,7 @@ class PM01FlatPPORunnerCfg(PM01RoughPPORunnerCfg):
     def __post_init__(self):
         super().__post_init__()
 
-        self.max_iterations = 1000
+        # self.max_iterations = 1000
         self.experiment_name = "pm01_flat"
         self.policy.actor_hidden_dims = [128, 128, 128]
         self.policy.critic_hidden_dims = [128, 128, 128]
diff --git a/source/pm01_dirct/pm01_dirct/tasks/direct/pm01_dirct/pm01_dirct_env_cfg.py b/source/pm01_dirct/pm01_dirct/tasks/direct/pm01_dirct/pm01_dirct_env_cfg.py
index dc40a51..e0f62b9 100644
--- a/source/pm01_dirct/pm01_dirct/tasks/direct/pm01_dirct/pm01_dirct_env_cfg.py
+++ b/source/pm01_dirct/pm01_dirct/tasks/direct/pm01_dirct/pm01_dirct_env_cfg.py
@@ -181,7 +181,7 @@ class Pm01DirctFaltEnvCfg(DirectRLEnvCfg):
     # joint_deviation_torso_reward_scale = -0.3
     # stand_still = -1.0
     track_lin_vel_xy_reward_scale = 1.0
-    track_ang_vel__reward_scale = 0.5
+    track_ang_vel__reward_scale = 1.0   #0.5
     lin_vel_z_l2_reward_scale = -2.0
     ang_vel_reward_scale = -0.05
     joint_torque_reward_scale = -2e-5